featurebyte/query_graph/transform/decompose_point.py

```diff
@@ -304,18 +388,43 @@ def update_ingest_graph_node_output_names(self, input_node_names: List[str]) ->
 
 
 class DecomposePointExtractor(
-    BaseGraphTransformer[DecomposePointGlobalState, DecomposePointGlobalState]
+    BaseGraphExtractor[DecomposePointState, BaseModel, DecomposePointState]
 ):
     """
     DecomposePointExtractor class is used to extract information about the decompose point
     used in offline ingest graph decomposition.
     """
 
-    def _compute(
+    def _pre_compute(
         self,
-        global_state: DecomposePointGlobalState,
+        branch_state: BaseModel,
+        global_state: DecomposePointState,
         node: Node,
-    ) -> None:
+        input_node_names: List[str],
+    ) -> Tuple[List[str], bool]:
+        skip_input_nodes = False
+        if isinstance(node.parameters, BaseGroupbyParameters):
+            # if groupby node has entity_ids, skip further exploration on input nodes
+            skip_input_nodes = True
+        return [] if skip_input_nodes else input_node_names, False
```

Comment:
This is an existing bug. We should not traverse further if the groupby node parameter is hit. This extractor is combined with `EntityExtractor` to have a consistent logic to extract entity related information from the graph. `EntityExtractor` class is removed in this PR as it is no longer used.